### PR TITLE
add check to verify existence of the tools of the workflow

### PR DESCRIPTION
**Pull request by gmauro**
I have added a check to verify that every tool in the workflow exists in the Galaxy instance

workflow_broken

without this check,  workflow.is_runnable return True

### CODE REVIEW

File: bioblend/galaxy/objects/wrappers.py [hunk 1]
```diff
@@ -183,6 +183,7 @@ class Workflow(Wrapper):
     def __init__(self, wf_dict, gi=None):
         super(Workflow, self).__init__(wf_dict, gi=gi)
         missing_ids = []
+        tools_list_by_id = [t.id for t in gi.tools.list()]
```
Comment (nsoranzo) [inline bioblend/galaxy/objects/wrappers.py]: Can you please `s/gi.tools.list()/gi.tools.get_previews()/` ?

Presently they are the same function, but in the future `list()` may be more expensive to evaluate.

**Comment by nsoranzo** [comment]
:+1: 

Thanks a lot for the fix, Mauro!

File: bioblend/galaxy/objects/wrappers.py (hunk 1 above)
Comment (nsoranzo) [inline bioblend/galaxy/objects/wrappers.py]: Also, this is breaking TraviCI testing, missing a `self.`?

```python
[t.id for t in self.gi.tools.get_previews()]
```

Comment (gmauro) [inline bioblend/galaxy/objects/wrappers.py]: No, the problems is in class TestWorkflow, that use a SAMPLE_WF_DICT without a galaxy instance.
class TestWorkflow(unittest.TestCase):

    def setUp(self):
        self.wf = wrappers.Workflow(SAMPLE_WF_DICT)

Comment (nsoranzo) [inline bioblend/galaxy/objects/wrappers.py]: Ah, of course, `self.` is not needed because this is the `__init__()` method and `gi` is also a parameter!

**Pull request merged**